[New Resources:] `azurerm_cdn_frontdoor_route`, `azurerm_cdn_frontdoor_custom_domain` and `azurerm_cdn_route_disable_link_to_default_domain` (#18231)
hashicorp/terraform-provider-azurerm: website/docs/r/cdn_frontdoor_route.html.markdown

---
subcategory: "CDN"
layout: "azurerm"
page_title: "Azure Resource Manager: azurerm_cdn_frontdoor_route"
description: |-
  Manages a Frontdoor Route.
---

# azurerm_cdn_frontdoor_route

> Manages a Frontdoor Route.

Reviewer: should these be "CDN FrontDoor" here for consistency across resources?

Author: Fixed.

  cdn_frontdoor_custom_domain_ids = [azurerm_cdn_frontdoor_custom_domain.contoso.id, azurerm_cdn_frontdoor_custom_domain.fabrikam.id]
  cdn_frontdoor_origin_ids        = [azurerm_cdn_frontdoor_origin.example.id]
  cdn_frontdoor_rule_set_ids      = [azurerm_cdn_frontdoor_rule_set.example.id]
  enabled                         = true

  forwarding_protocol            = "HttpsOnly"
  https_redirect_enabled         = true
  link_to_default_domain_enabled = false
  patterns_to_match              = ["/*"]
  supported_protocols            = ["Http", "Https"]

  cache {
    query_string_caching_behavior = "IgnoreSpecifiedQueryStrings"
    query_strings                 = ["account", "settings"]
    compression_enabled           = true
    content_types_to_compress     = ["text/html", "text/javascript", "text/xml"]
  }
}
```

## Arguments Reference

* `name` - (Required) The name which should be used for this Frontdoor Route. Valid values must begin with a letter or number, end with a letter or number and may only contain letters, numbers and hypens with a maximum length of 90 characters. Changing this forces a new Frontdoor Route to be created.

> * `cdn_frontdoor_endpoint_id` - (Required) The ID of the Frontdoor Route. Changing this forces a new Frontdoor Route to be created.

Reviewer: ```suggestion
* `cdn_frontdoor_endpoint_id` - (Required) The ID of the CDN FrontDoor Endpoint within which this CDN FrontDoor Route should exist. Changing this forces a new Frontdoor Route to be created.
```

Author: Fixed.

> * `cdn_frontdoor_origin_group_id` - (Required) The resource ID of the Frontdoor Origin Group.

Reviewer: does this want to be?

```suggestion
* `cdn_frontdoor_origin_group_id` - (Required) The ID of the CDN FrontDoor Origin Group within which this CDN FrontDoor Route should be created.
```

Author: Fixed.

* `cdn_frontdoor_origin_ids` - (Required) One or more Frontdoor Origin resource IDs that this Frontdoor Route will link to.

* `supported_protocols` - (Required) One or more Protocols supported by this Frontdoor Route. Possible values are `Http` or `Https`.

~> **NOTE:** If `https_redirect_enabled` is set to `true` the `supported_protocols` field must contain both `Http` and `Https` values.

> * `patterns_to_match` - (Reqired) The route patterns of the rule.

Reviewer: ```suggestion
* `patterns_to_match` - (Required) The route patterns of the rule.
```

Author: Fixed.

> * `cdn_frontdoor_custom_domain_ids` - (Optional) One or more resource IDs of the Frontdoor Custom Domains to associate with the Frontdoor Route.

Reviewer: ```suggestion
* `cdn_frontdoor_custom_domain_ids` - (Optional) A list of CDN FrontDoor Custom Domain IDs to associate with this CDN FrontDoor Route.
```

Author: Fixed.

* `link_to_default_domain_enabled` - (Optional) Will the Frontdoor Route be linked to the default domain endpoint? Possible values are `true` or `false`. Defaults to `false`.

* `cache` - (Optional) A `cache` block as defined below.

~> **NOTE:** To to disable caching, do not provide the `cache` block in the configuration file.

* `enabled` - (Optional) Is this Frontdoor Route enabled? Possible values are `true` or `false`. Defaults to `true`.

* `forwarding_protocol` - (Required) The Protocol that will be use when forwarding traffic to backends. Possible values are `HttpOnly`, `HttpsOnly` or `MatchRequest`.

* `https_redirect_enabled` - (Optional) Automatically redirect HTTP traffic to HTTPS traffic? Possible values are `true` or `false`. Defaults to `true`.

~> **NOTE:** The `https_redirect_enabled` rule is the first rule that will be executed.

> * `cdn_frontdoor_origin_path` - (Optional) A directory path on the origin that Frontdoor can use to retrieve content from(e.g. contoso.cloudapp.net/originpath).

Reviewer: ```suggestion
* `cdn_frontdoor_origin_path` - (Optional) A directory path on the origin that Frontdoor can use to retrieve content from (e.g. `contoso.cloudapp.net/originpath`).
```

Author: Fixed.

> * `cdn_frontdoor_rule_set_ids` - (Optional) One or more Frontdoor Rule Set Resource ID's.

Reviewer: ```suggestion
* `cdn_frontdoor_rule_set_ids` - (Optional) A list of the CDN FrontDoor Rule Set IDs which should be assigned to this CDN FrontDoor Route.
```

Author: Fixed.
